Ewald.js

```diff
@@ -0,0 +1,120 @@
+// We zouden de bankCode natuurlijk ook gewoon in de <option>s van de <select> kunnen plaatsen..
+// maar dan hoef je niet te oefenen met arrays :)
+function bepaalBankCode(bank) {
+    var bankCode = '';
+    for(var i=0; i < bankIdentificatie.length; i++){
+    	if(bankIdentificatie[i][0]===bank){
+				bankCode = bankIdentificatie[i][1];
+      }
+    }
+    // haal hier de bankCode op uit 'bankIdentificatie'
+
+    return bankCode;
+}
+
+// Maak van het 7- of 9-cijferige nummer een 10-cijferig rekeningnummer
+function bepaalHeleRekeningNummer(rekeningNummer) {
+
+    // zorg er hier voor dat het rekeningnummer altijd 10 cijfers lang is, eventueel met nullen ervoor
+    while(rekeningNummer.length < 10){
+      rekeningNummer = '0'+rekeningNummer;
+    }
+
+    while(rekeningNummer.length > 10){
+      rekeningNummer = rekeningNummer.slice(0, 10);
+    }
+
+    return rekeningNummer;
+}
+
+// bereken nu het controlegetal
+//    Neem de rekeningidentificatie, plaats de landcode daarachter
+//    Vervang alle letters door twee cijfers gebaseerd op de volgorde in het alfabet (A=10, B=11, C=12, ..., Z=35)
+//    Voeg twee nullen toe aan het einde
+//    Deel dit door 97 en neem de rest (mod97)
+//    Het controlegetal is nu 98 - <deze rest>
+//    Is het controlegetal kleiner dan 10? Zet er dan een 0 voor. (2 wordt dus 02)
+//    Zie http://nl.wikipedia.org/wiki/International_Bank_Account_Number#Controlegetal
+function bepaalControleGetal(bankCode, landCode, rekeningNummer) {
+    var controleGetal = 0;
+    var nummerAlsString = bankCode + rekeningNummer + landCode;
+    // In de geconverteerdeString zijn letters vervangen door hun getalwaarde
+    // Hierbij moet A de waarde 10 krijgen, B wordt 11, C = 12, etc)
+    var geconverteerdeString = '';
+
+    // Een simpele manier om dit te doen is b.v. door elk karakter in nummerAlsString te bekijken en
+    // te controleren of dit een getal is of niet (met isNaN: isNaN(3) = false, isNaN('d') = true).
+    // Vervolgens kun je getallen direct toevoegen aan de geconverteerde string, letters moet je eerst omzetten
+    // voordat je ze toe kunt voegen.
+    // Hint: 'ABCDEFG'.charCodeAt(2) - 'A'.charCodeAt(0) geeft als resultaat 2, want 'C' - 'A' = 67 - 65 = 2;
+    // A moet 10 worden, B = 11, C = 12, etc.
+
+    for(var i=0;i<nummerAlsString.length;i++){
+      var letter = nummerAlsString.charAt(i);
+      if(isNaN(nummerAlsString.charAt(i))){
+        letter = letter.charCodeAt(0)-'A'.charCodeAt(0)+10;
+      }
+      geconverteerdeString = geconverteerdeString+letter
+    }
+
+    // Voeg nu twee nullen toe aan het einde en gebruik mod97 i.p.v. %97
+    geconverteerdeString = mod97(geconverteerdeString+'00');
+    // bepaal het controlegetal en return dit (98 - het resultaat van de mod97, altijd twee cijfers)
+    controleGetal = 98 - geconverteerdeString;
+
+    if(controleGetal.toString().length == 1) {
+      controleGetal = '0' + controleGetal;
+    }
+    return controleGetal;

```

Comment:
Dit werkt prima, maar het is goed om even stil te staan bij wat hier gebeurt.
op regel 100 was controleGetal een integer, maar als het maar 1 teken bevatte, werd het op regel 103 een string.
Deze functie retourneert dus een integer of een string, en je weet bij voorbaat niet wat. 

Dat is gevaarlijk: op regel 34 ga je vervolgens verschillende strings bij elkaar voegen met de +-operator. Maar als dat allemaal variabelen waren, die zowel string als integer kunnen zijn, dan had het zomaar gekund dat je was gaan optellen in plaats van samenvoegen.

JavaScript is losely typed, en maakt hier dus geen probleem van. Het is aan de ontwikkelaar om dit soort gevaren af te vangen. In het algemeen is het aan te raden om met functies als parseInt() en toString() er voor te zorgen dat een functie altijd het type retourneert dat je verwacht.